core/client/src/runtime_api.rs

```diff
@@ -139,6 +139,7 @@ decl_runtime_apis! {
 	pub trait TaggedTransactionQueue {
 		/// Validate the given transaction.
 		fn validate_transaction(tx: <Block as BlockT>::Extrinsic) -> TransactionValidity;
+		fn get_account_nonce(account_id: &sr25519::Public) -> u64;
```

Comment:
(i would do the nonce-fetching on a per-runtime basis, but we have designed the abstraction of `TaggedTransactionQueue` specifically to avoid nonces in this layer)